adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/HttpClientBuilder.java

```diff
@@ -287,16 +289,20 @@ public HttpClient build() {
                 cm = new SingleClientConnManager(registry);
             }
             BasicHttpParams params = new BasicHttpParams();
-            if (socketTimeout > -1)
-            {
+
+            if (proxyHost != null) {
+                params.setParameter(ConnRoutePNames.DEFAULT_PROXY, proxyHost);
```

Comment:
Looks like this is deprecated - https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/conn/params/ConnRoutePNames.html#DEFAULT_PROXY